README.md

```diff
@@ -1 +1,99 @@
 # Unikraft "hello-world" Application with Coverage Support (gcov)
+
+This builds the Unikraft hello-world image with coverage support used by [gcov](https://gcc.gnu.org/onlinedocs/gcc/Gcov.html).
+The actual support is provided in gcc via the `-fprofile-arcs` option.
+Coverage support can be used for profile-guided optimization (PGO).
+This is why the usual way to compile and link a program under test is via the `-fprofile-generate` option; it includes the `-fprofile-arcs` option.
+
+## Dependencies
+
+The following libraries / dependencies are required:
+* `lib-pthread-embedded`
+* `lib-libunwind`
+* `lib-compiler-rt`
+* `lib-libcxxabi`
+* `lib-libcxx`
+* `lib-newlib`
+* `lib-gcc`
+
+For `lib-gcc` use the [UPB clone](https://github.com/cs-pub-ro/lib-gcc), the `gcov` branch.
+Similarly, for Unikraft use [UPB clone](https://github.com/cs-pub-ro/unikraft), the `gcov` branch.
+
+For the other libraries, use the `staging` branch.
+
+The dependencies have to be included in that order as done in the `Makefile`.
+Update the `Makefile` variables (`UK_ROOT`, `UK_LIBS`, `LIBS`) according to your setup.
+
+## Configuration
+
+Enter the configuration screen:
+```
+make menuconfig
+```
+Enable the following configuration options:
+* `Platform Configuration` -> `KVM (guest)`
+* `Library Configuration`
+  * `libpthread-embedded`
+  * `libunwind`
+  * `compiler-rt`
+  * `libcxxabi`
+  * `libcxx`
+  * `newlib`
+  * `gcc`
+    * `libgcov: A Profiling Library`
+  * `vfscore: Configuration`
+    * `Automatically mount a root filesystem (/)`
+    * `Default root filesystem`: Choose `9pfs`
+    * `Default root device`: Choose `fs0`
+  * `POSIX sysinfo: Information about system parameters`
+* `Build Options`
+  * `PGO - Profile-guided optimization`
```

Comment:
I took the liberty to write the needed `Config.uk` content. One option is not working properly and was commented out. I could not find how to automatically assign strings.

The configuration:
```
menuconfig APPHELLOWORLDGCOV_DEPENDENCIES
        bool "Application to showcase gcov usage"
        default y
        select PLAT_KVM

        select LIBPTHREAD_EMBEDDED
        select LIBUNWIND
        select LIBCOMPILER_RT
        select LIBCXXABI
        select LIBCXX
        select LIBNEWLIBC
        select LIBGCC
        select LIBGCOV
        select LIBPOSIX_SYSINFO
        select LIBVFSCORE_AUTOMOUNT_ROOTFS
        select OPTIMIZE_PGO

choice LIBVFSCORE_ROOTFS
        default LIBVFSCORE_ROOTFS_9PFS
        #TODO NOT WORKING# LIBVFSCORE_ROOTDEV=fs0
endchoice
```

Reply:
I tried creating a Config.uk as well, only to be stopped by the same issue. Unfortunately, from what I have read, that would not be possible due to how kconfigs work: [see here](https://github.com/torvalds/linux/blob/d8a5b80568a9cb66810e75b182018e9edb68e8ff/Documentation/kbuild/kconfig-language.txt#L67). As such, the Config.uk from vfscore overrides any default value we would attempt to impose on that symbol. I am afraid that changing the default value from vfscore's Config.uk is the only way we could do that, since it is the first defined default value.